### ISSUE TITLE
nested functions and macros aren't picked up

### ISSUE THREAD

**Issue by expipiplus1** [BlankSpruce/gersemi#39]
This for examples warns about an unknown function `bar`

```cmake
function(foo)
    macro(bar)
    endmacro()
    bar()
endfunction()
```

**Comment by BlankSpruce** [comment]
Is there a repository that I could clone and experiment with that use-case? I'd like to evaluate whether it's worth to interpret these nested commands. I'm afraid it could lead to warnings like [shown here](https://github.com/BlankSpruce/gersemi/issues/11#issuecomment-1768891385) when those nested definitions reuse names:

```cmake
function(foo1)
    macro(bar)
    endmacro()
    bar()
endfunction()

function(foo2)
    macro(bar)
    endmacro()
    bar()
endfunction()
```

**Comment by expipiplus1** [comment]
Sure check out https://github.com/shader-slang/slang there's a script calling gersemi here: [shader-slang/slang@`f1da05f` (#5409)](https://github.com/shader-slang/slang/pull/5409/commits/f1da05fa01885c66f9c1434a694f3f1a530fabd2)

**Comment by BlankSpruce** [comment]
Fixed in 0.17.0.

I've noticed in [extras/formatting.sh](https://github.com/shader-slang/slang/pull/5409/commits/f1da05fa01885c66f9c1434a694f3f1a530fabd2#diff-5f736ca33087c74dd150e6dc72f5adf26eb2db97562aead803ad6d12f2646db7) that you're listing CMake files with `git ls-files` but it's probably not necessary because gersemi was always capable to search these patterns in passed directories itself. I've extended `--help` so it's a little bit more obvious.
